test/units/modules/network/vyos/test_vyos_firewall_interfaces.py

```diff
@@ -0,0 +1,296 @@
+            "set interfaces ethernet eth1 firewall local ipv6-name 'V6-LOCAL'",
+            "set interfaces ethernet eth3 firewall in name 'INBOUND'",
+            "set interfaces ethernet eth3 firewall out name 'OUTBOUND'",
+            "set interfaces ethernet eth3 firewall local name 'LOCAL'",
+            "set interfaces ethernet eth3 firewall local ipv6-name 'V6-LOCAL'"
+        ]
+        self.execute_module(changed=True, commands=commands)
+
+    def test_vyos_firewall_rule_set_02_merged_idem(self):
+        set_module_args(
+            dict(config=[
+                dict(name='eth0',
+                     access_rules=[
+                         dict(afi='ipv4',
+                              rules=[
+                                  dict(name='INBOUND', direction='in'),
+                                  dict(name='OUTBOUND', direction='out'),
+                                  dict(name='LOCAL', direction='local')
+                              ]),
+                         dict(afi='ipv6',
+                              rules=[dict(name='V6-LOCAL', direction='local')])
+                     ]),
+                dict(name='eth2',
+                     access_rules=[
+                         dict(afi='ipv4',
+                              rules=[
+                                  dict(name='INBOUND', direction='in'),
+                                  dict(name='OUTBOUND', direction='out'),
+                                  dict(name='LOCAL', direction='local')
+                              ]),
+                         dict(afi='ipv6',
+                              rules=[dict(name='V6-LOCAL', direction='local')])
+                     ])
+            ],
+                state="merged"))
+        self.execute_module(changed=False, commands=[])
+
+    def test_vyos_firewall_rule_set_01_deleted(self):
+        set_module_args(
+            dict(config=[
+                dict(name='eth0',
+                     access_rules=[dict(afi='ipv4'),
+                                   dict(afi='ipv6')])
+            ],
+                state="deleted"))
+        commands = [
+            "delete interfaces ethernet eth0 firewall in name",
+            "delete interfaces ethernet eth0 firewall local name",
+            "delete interfaces ethernet eth0 firewall out name",
+            "delete interfaces ethernet eth0 firewall local ipv6-name"
+        ]
+        self.execute_module(changed=True, commands=commands)
+
+    def test_vyos_firewall_rule_set_03_deleted(self):
+        set_module_args(
+            dict(config=[dict(name='eth0'),
+                         dict(name='eth2')],
+                 state="deleted"))
+        commands = [
+            "delete interfaces ethernet eth0 firewall",
+            "delete interfaces ethernet eth2 firewall"
+        ]
+        self.execute_module(changed=True, commands=commands)
+
+    def test_vyos_firewall_rule_set_03_deleted(self):
```

Comment:
Can a comment be added or the name of the testcase be changed, to indicate, what level of granularity of deletion, is tested here ?

Reply:
Done: I have modified the name of the function/test case keeping the granularity in mind, rest of the scenarios have been taken care in Integration tests, I will cover the remaining in separate PR